js/main.js

```diff
@@ -184,56 +188,81 @@ function saveVerse(event) {
-    } else { // if this is not a new entry or you're editing...
-      var currentDiv = event.target.closest('li').firstElementChild; // grab the closest li element and its first child which should be a div
-      var currentVerse = currentDiv.firstElementChild.textContent; // grab the first child of that div which should be a p element and get its text content
-      var $currentVerse = document.getElementById('journal-verse'); // grab the p elemennt that is used to show the verse
-      $currentVerse.textContent = currentVerse; // set the p element that shows the verse to the current verse you decided to journal on
-
-      if (data.entries[getEntryIndex()].entry === undefined) {
-        $textArea.value = '';
-      } else {
-        $textArea.value = data.entries[getEntryIndex()].entry;
-      }
+  if (event.target.textContent === 'Edit' || event.target.textContent === 'Write Journal Entry') { // if the user clicks the button and the text content says 'Edit'...
+    var currentDiv = event.target.closest('li').firstElementChild; // grab the closest li element and its first child which should be a div that is a parent of the element which holds the verse
+    var currentVerse = currentDiv.firstElementChild.textContent; // grab the first child of that div which should be a p element and get its text content
+    data.editing = event.target.closest('li'); // set the editing property of our data model to the event.target which is the button element that you clicked so we can access the id
+    var $ul = document.querySelector('#journal-entries'); // grab the ul element so we can append the verse that's been selected
+    var $textArea = document.querySelector('textarea'); // grab the textarea element so we can use to update the textarea box
+    $ul.className = 'padding-bottom'; // add some padding to the bottom of $ul so we can add some spacing between the verse & textarea
+
+    data.view = 'edit-page'; // set property of view to 'edit-page' in the data model so we can change views
+    viewSwapper(); // change the page view
+
+    if ($ul.hasChildNodes() === false) { // if this is the first new entry...
+      var $li = document.createElement('li'); // create a new 'li' element
+      var $p = document.createElement('p'); // create a new 'p' element
+
+      $p.id = 'journal-verse'; // set the id of 'p' to 'journal-verse'
+      $p.textContent = data.editing.firstChild.textContent; // get the verse in the data entry and set it as the shown verse
+
+      $ul.appendChild($li); // append $li to $ul
+      $li.appendChild($p); // append $p to $li
+    } else { // if this is not a new entry aka you're editing...
+      var $editPageVerse = document.getElementById('journal-verse'); // grab the p element that is used to show the verse
+      $editPageVerse.textContent = currentVerse; // set the p element that shows the current verse you decided to journal in the edit page
+      $textArea.value = data.editing.lastChild.firstChild.textContent;
```

Comment:
Instead of doing this, I would use `querySelector` for a class or something, you don't want your JavaScript dependent on a specific order of elements in the HTML.